lib/config/generator.js

```diff
@@ -0,0 +1,222 @@
+
+            if (!violatedRuleCount) { return this._config; }
+
+            console.log(_path + ' violates ' + violatedRuleCount + ' rule' + (violatedRuleCount > 1 ? 's' : ''));
+
+            var errorPrompts = generateRuleHandlingPrompts(errorList);
+
+            // TODO: Extract method?
+            return this._getUserViolationChoices(errorPrompts)
+            .then(function handleViolatedRules(choices) {
+                errorPrompts.forEach(function(errorPrompt, idx) {
+                    var associatedRuleName = errorList[idx].rule;
+                    var userChoice = choices[errorPrompt];
+
+                    _this._processViolationChoice(userChoice, associatedRuleName);
+                });
+
+                return this._config;
+            });
+        }.bind(this))
+        .then(function flushConfig(config) {
+            fs.writeFileSync(process.cwd() + '/.jscsrc', JSON.stringify(config, null, '\t'));
+            console.log('Generated a .jscsrc configuration file in ' + process.cwd());
+        });
+    },
+
+    /**
+     * @private
+     * @param  {String[]} presetNames
+     * @param  {Object[]} errorsCollection
+     */
+    _showErrorCounts: function(presetNames, errorsCollection) {
+        var errorCounts = errorsCollection.map(function(error) {
+            return error.getErrorCount();
+        });
+
+        var table = getTable();
+
+        presetNames.forEach(function(name, idx) {
+            table.push([idx + 1, name, errorCounts[idx]]);
+        });
+
+        console.log(table.toString());
+    },
+
+    /**
+     * Prompts the user to choose a preset
+     *
+     * @private
+     * @param {Object} prompt
+     * @return {Promise}
+     */
+    _getUserPresetChoice: function(prompt) {
+        return this._showPrompt(prompt);
+    },
+
+    /**
+     * Prompts the user to nullify rules or fix violations themselves
+     *
+     * @private
+     * @param  {Object[]} errorPrompts
+     * @return {Promise}
+     */
+    _getUserViolationChoices: function(errorPrompts) {
+        return this._showPrompt(errorPrompts);
+    },
+
+    /** @private */
+    _showPrompt: vowNode.promisify(prompt.get.bind(prompt)),
+
+    /**
+     * @private
+     * @param  {String} choice
+     * @param  {String} ruleName
+     */
+    _processViolationChoice: function(choice, ruleName) {
+        var lower = choice.toLowerCase();
+
+        if (lower === 'e') {
+            this._config[ruleName] = null;
+        }
+    },
+
+    /**
+     * @private
+     * @param  {String} path
+     * @param  {String} presetName
+     * @return {Promise}
+     */
+    _checkAgainstPreset: function(path, presetName) {
+        var checker = getChecker();
+
+        checker.configure({preset: presetName});
+
+        return checker.checkPath(path);
+    }
+};
+
+function getChecker() {
+    var checker = new Checker();
+    checker.registerDefaultRules();
+    return checker;
+}
+
+function normalizePath(filepath, basePath) {
```

Comment:
This should be extracted to utils too. This same "if relative, then resolve" code is in 4 spots.